receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -144,7 +144,9 @@ func (s *scraper) getProcessMetadata() ([]*processMetadata, error) {
 
 		executable, err := getProcessExecutable(handle)
 		if err != nil {
-			errs.AddPartial(1, fmt.Errorf("error reading process name for pid %v: %w", pid, err))
+			if !s.config.MuteProcessNameError {
+				errs.AddPartial(1, fmt.Errorf("error reading process name for pid %v: %w", pid, err))
+			}
```

Comment:
This is catching file access errors at the first function only. But others like `command, err := getProcessCommand` that called below can cause that as well. So if for example the order is changed it'll fire anyway.

Reply:
The error that this PR is aiming to suppress only occurs when `process.Exe()` is called; it is an error that occurs with `os.Readlink`. As far as I have tested, the error that this config is hoping to mute only happens on `getProcessExecutable`, which is the only place that calls `process.Exe()`.
If we keep this solution, the config could use renaming; `MuteReadlinkError` is more descriptive of the goal of this flag.

Reply:
>If we keep this solution, the config could use renaming; MuteReadlinkError is more descriptive of the goal of this flag.

Coming back to this, I've realized that's actually not a good name for this config. On Windows, this doesn't have a "readlink error" per se, it is different types of errors (though still occurring when trying to get Executable). I think the current name still makes the most sense since it covers what a user on either platform expects to mute with this flag.